js/interface/marker-panel.js

```diff
@@ -66,14 +70,18 @@ Fliplet.InteractiveMap.component('marker-panel', {
       })
 
       window.iconPickerProvider.then((data) => {
-        if (data.data) {
-          this.icon = data.data.icon
+        if (!data.data.icon) {
+          this.emptyIconNotification = true;
+        } else {
+          this.icon = data.data.icon;
+          this.emptyIconNotification = false;
         }
-        this.onInputData()
-        window.iconPickerProvider = null
-        Fliplet.Studio.emit('widget-save-label-reset')
-        return Promise.resolve()
-      })
+
+        this.onInputData();
+        window.iconPickerProvider = null;
+        Fliplet.Studio.emit('widget-save-label-reset');
+        return Promise.resolve();
```

Comment:
We don't add `;` at the end of the line. @squallstar Can you explain the reason and the scenarios under which we skip the `;` characters?

Reply:
`;` should not be added on code that gets compiled and transpiled via Gulp, because the transpiler takes care of adding it hence our source code is cleaner and simpler to write and maintain. Same applies for ES6+ features, e.g. `async/await`, `const` and `arrow functions`.

At this time we only have a few components whose code gets transpiled:

- Interactive graphics (this component)
- Theme Appearance 2.0
- Fliplet Studio (Upplabs have no access to this)

Reply:
@squallstar  Got it. Should I fix it?

Reply:
@MaksymShokin if your editor support ESLint (or JSHint), this repository is set up with a file to automatically fix this for you and warn you about problems: https://github.com/Fliplet/fliplet-widget-interactive-map/blob/master/.eslintrc (and the corresponding .jslintrc)